uninstall

```diff
@@ -0,0 +1,226 @@
+#!/System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/ruby
+
+require "fileutils"
+require "optparse"
+require "pathname"
+
+# Default options
+options = {
+  :force => false,
+  :quiet => false,
+  :dry_run => false,
+  :skip_cache_log => false,
+}
+
+module Tty extend self
+  def blue; bold 34; end
+  def white; bold 39; end
+  def red; underline 31; end
+  def reset; escape 0; end
+  def bold n; escape "1;#{n}" end
+  def underline n; escape "4;#{n}" end
+  def escape n; "\033[#{n}m" if STDOUT.tty? end
+end
+
+class Array
+  def shell_s
+    cp = dup
+    first = cp.shift
+    cp.map{ |arg| arg.gsub " ", "\\ " }.unshift(first) * " "
+  end
+end
+
+class Pathname
+  def subdirs
+    children.select { |child| child.directory? }
+  end
+
+  def resolved_path
+    self.symlink? ? dirname+readlink : self
+  end
+
+  def /(other)
+    self + other.to_s
+  end
+
+  def pretty_print
+    if self.symlink?
+      puts to_s + " -> " + resolved_path.to_s
+    elsif self.directory?
+      puts to_s + "/"
+    else
+      puts to_s
+    end
+  end
+end
+
+def ohai *args
+  puts "#{Tty.blue}==>#{Tty.white} #{args.shell_s}#{Tty.reset}"
+end
+
+def warn warning
+  puts "#{Tty.red}Warning#{Tty.reset}: #{warning.chomp}"
+end
+
+def system *args
+  warn "Failed during: #{args.shell_s}" unless Kernel.system(*args)
+end
+
+####################################################################### script
+
+$HOMEBREW_PREFIX_CANDIDATES=[]
+
+OptionParser.new do |opts|
+  opts.banner = "Homebrew Uninstaller\nUsage: ./uninstall [options]"
+  opts.summary_width = 16
+  opts.on("-pPATH", "--path=PATH", "Set Homebrew prefix, like /usr/local.") { |p| $HOMEBREW_PREFIX_CANDIDATES << Pathname.new(p) }
+  opts.on("--skip-cache-log", "Skips removal of Caches and Logs files.") { |p| options[:skip_cache_log] = true }
```

Comment:
Ah, actually, `Skips removal of HOMEBREW_CACHE and HOMEBREW_LOGS` and make it `--skip-cache-and-logs`.